helsingborg-stad/helsingborg-io-sls-api
# Feat/metrics-endpoint

## Explain the changes you’ve made

Added metrics api endpoint with ekb response (stubbed for now - data 
fetching to be implemented in other task).

NOTE: also requires gateway resources in [resources repo PR 111](https://github.com/helsingborg-stad/helsingborg-io-sls-resources/pull/111). 

Also added new custom error `InvalidArgumentError` to correctly return 400 on invalid inputs.

## Explain why these changes are made

To enable metrics to be exposed for consumption by Prometheus and visualized with Grafana for our organization-wide statistics dashboards.

## How to test

Concrete example:

1. Checkout this branch
2. Run tests
3. Check that endpoint `/metrics/ekb` requires API key
4. Check that endpoint `/metrics/ekb` requires a `?format` query parameter, and only `ebnf` is valid
5. Check that it returns 200 with empty body with all correct parameters
